release_notes/model.py

```diff
@@ -415,54 +415,19 @@ def _source_component(
         return None
 
 
-def create_release_notes_obj(
-    component_descriptor_lookup,
-    version_lookup,
+def create_release_notes_entry(
     source_block: SourceBlock,
-    source_component: ocm_model.Component,
-    current_component: ocm_model.Component,
-) -> ReleaseNote:
-    target = source_block.source
-    if isinstance(target, git.Commit):
-        ref = create_commit_ref(target, source_block)
-    elif isinstance(target, github3.pulls.ShortPullRequest):
-        ref = create_pull_request_ref(target, source_block)
-    else:
-        raise NotImplementedError(
-            f"Release note creation not implemented for target-type {type(target)}"
-        )
-
-    author = author_from_source(target)
-
-    if source_block.component_name:
-        source_component = _source_component(
-            component_descriptor_lookup=component_descriptor_lookup,
-            version_lookup=version_lookup,
-            current_component=current_component,
-            source_component_name=source_block.component_name,
-        )
-
-    # access
-    source_component_access = ocm.util.main_source(
-        component=source_component,
-        no_source_ok=False,
-    ).access
-
-    current_src_access = ocm.util.main_source(
-        component=current_component,
-        no_source_ok=False,
-    ).access
-
-    from_same_github_instance = current_src_access.hostname() in source_component_access.hostname()
-    is_current_repo = current_component.name == source_component.name
-
-    return ReleaseNote(
-        source_block=source_block,
-        author=author,
-        reference=ref,
-        source_component=source_component,
-        is_current_repo=is_current_repo,
-        from_same_github_instance=from_same_github_instance
+) -> ReleaseNoteEntry:
+    return ReleaseNoteEntry(
```

Comment:
since this function now almost exclusively wraps the initialiser, how about, well, using the initialiser (inline)? As an alternative, you might mv this either to SourceBlock (`as_release_note_entry`), or as a factory-method to RNE

wdyt?